generators/client/templates/angular/src/main/webapp/app/blocks/interceptor/_notification.interceptor.ts

```diff
@@ -16,50 +16,51 @@
  limitations under the License.
 -%>
-import { JhiAlertService, JhiHttpInterceptor } from 'ng-jhipster';
-import { RequestOptionsArgs, Response } from '@angular/http';
+import { JhiAlertService } from 'ng-jhipster';
+import { HttpInterceptor, HttpRequest, HttpErrorResponse, HttpResponse, HttpHandler, HttpEvent } from '@angular/common/http';
 import { Injector } from '@angular/core';
 import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/do';
 
-export class NotificationInterceptor extends JhiHttpInterceptor {
+export class NotificationInterceptor implements HttpInterceptor {
 
     private alertService: JhiAlertService;
 
     // tslint:disable-next-line: no-unused-variable
     constructor(private injector: Injector) {
-        super();
         setTimeout(() => this.alertService = injector.get(JhiAlertService));
     }
 
-    requestIntercept(options?: RequestOptionsArgs): RequestOptionsArgs {
-        return options;
-    }
-
-    responseIntercept(observable: Observable<Response>): Observable<Response> {
-        return observable.map((response: Response) => {
-            const headers = [];
-            response.headers.forEach((value, name) => {
-                if (name.toLowerCase().endsWith('app-alert') || name.toLowerCase().endsWith('app-params')) {
-                    headers.push(name);
+    intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        return next.handle(request).do((event: HttpEvent<any>) => {
+            if (event instanceof HttpResponse) {
+                const arr = event.headers.keys();
+                let alert = null;
+                let alertParams = null;
+                for (let i = 0; i < arr.length; i++) {
+                    if (arr[i].endsWith('app-alert')) {
+                        alert = event.headers.get(arr[i]);
+                    } else if (arr[i].endsWith('app-params')) {
+                        alertParams = event.headers.get(arr[i]);
+                    }
                 }
-            });
-            if (headers.length > 1) {
-                headers.sort();
-                const alertKey = response.headers.get(headers[ 0 ]);
-                if (typeof alertKey === 'string') {
-                    if (this.alertService) {
-                        <%_ if (enableTranslation) { _%>
-                        const alertParam = headers.length >= 2 ? response.headers.get(headers[ 1 ]) : null;
-                        this.alertService.success(alertKey, { param : alertParam }, null);
-                        <%_ } else { _%>
-                        this.alertService.success(alertKey, null, null);
-                        <%_ } _%>
+                if (alert) {
+                    if (typeof alert === 'string') {
+                        if (this.alertService) {
+                            <%_ if (enableTranslation) { _%>
+                                const alertParam = alertParams;
```

Comment:
indentation of rendered code seems wrong, template indentation should be independent from code indent